PULSEBS-95

LeaX-XIV opened the pull request:
- Extended GET /api/users/{userId}/lectures to allow uers of type booking manager to access the list of all (non-cancelled) lectures.

- Moved the function serving the API endpoint in a new file, and refactored the function, lowering its cognitive complexity.

LeaX-XIV committed:
Extend endpoint

[PULSEBS-95] Allow users of type booking manager to access the list
 of all lectures.
[PULSEBS-102] Refactored endpoint to lower cognitive complexity.

LeaX-XIV committed:
Add test case [PULSEBS-65]

Test case to check for validity of call as a booking manager.
Update test reports.

Francexi commented on server/API/ListLectures.php:
```diff
@@ -0,0 +1,270 @@
+		$endTS = 9e9;
+
+		if ($startISODate) {
+			$Ymd = explode('-', $startISODate);
```
why don't you use DateTime? Like, for example $Ymd = new DateTime($startISODate); $startTS = $Ymd->getTimestamp;

Francexi commented on server/API/ListLectures.php:
```diff
@@ -0,0 +1,270 @@
+		C.ID as courseId,
+		C.name as courseName,
+		L.start_ts as startTS,
+		L.end_ts as endTS,
```
shouldn't the endpoint also give the year and semester information about the course?

LeaX-XIV committed:
Add new info to returned body

Year and semester of the course.
DB does not support current versiones of queries -> placeholder to
remove once in master branch.
